### PR TITLE
Re-write the greentea_send_kv() functions to replace the use of printf,

### PR DESCRIPTION
**Pull request by adbridge**
## Description

with a set of underlying functions to write the preamble, postamble,
strings and integer values, directly to the serial port (for target to host comms)
using the Rawserial class. Raw serial can be used safely from
within an interrupt context, whereas printf cannot. Thus this change
allows key/values to be written to the host from within interrupt context.

Added instantiation of the Rawserial class, greentea_serial
Added new functions:
- `greentea_write_preamble()`
- `greentea_write_postamble()`
- `greentea_write_string()`
- `greentea_write_int()`

This is an internal change only within greentea-client and does not affect
the user use of `greentea_send_kv()`.

### CODE REVIEW

**Comment by adbridge** [comment]
@PrzemekWirkus @mazimkhan please review

File: source/test_env.cpp [hunk 1]
```diff
@@ -172,52 +179,138 @@ void greentea_notify_coverage_end() {
...
+ *        to the host. The integer value is converted to a string and 
+ *        and then written character by character directly to the serial 
+ *        port using the greentea_serial, Rawserial object.
+ *        sprintf() is used to convert the int to a string. Sprintf if
+ *        inherently thread safe so can be used.
+ *
+ * \param val - integer value
+ *
+ */
+#define MAX_INT_STRING_LEN 15
```
Comment (PrzemekWirkus) [inline source/test_env.cpp:250]:
``` c++
const int MAX_INT_STRING_LEN = 16;
```

**Comment by PrzemekWirkus** [comment]
+1

**Comment by PrzemekWirkus** [comment]
@0xc0170  Can you check if we are using `RAWSerial` "the right way"?

**Comment by PrzemekWirkus** [comment]
@0xc0170 I will merge this PR but if you have any doubts, please let me know :)

**Pull request merged**